include/deal.II/numerics/data_out_dof_data.h

```diff
@@ -1094,12 +1113,17 @@ DataOut_DoFData<DoFHandlerType, patch_dim, patch_space_dim>::add_data_vector(
 
 
 
-template <typename DoFHandlerType, int patch_dim, int patch_space_dim>
+template <typename DoFHandlerType,
+          int patch_dim,
+          int patch_space_dim,
+          typename Number>
 template <typename VectorType>
 void
-DataOut_DoFData<DoFHandlerType, patch_dim, patch_space_dim>::add_data_vector(
-  const VectorType &                                        vec,
-  const DataPostprocessor<DoFHandlerType::space_dimension> &data_postprocessor)
+DataOut_DoFData<DoFHandlerType, patch_dim, patch_space_dim, Number>::
+  add_data_vector(const VectorType &vec,
+                  const DataPostprocessor<DoFHandlerType::space_dimension,
+                                          typename VectorType::value_type>
+                    &data_postprocessor)
```

Comment:
I made the type of `DataPostprocessor` equal to the type of the `Vector`. But I have the feeling that you do a conversion. For example `Vector` can be `float` and you convert it to `double`. This wouldn't work for `std::complex`

Why you convert `float` to `double`? It is possible to store `float` in the vtk/vtu file, there is no need for a conversion.